### PR TITLE
Animation Indicator

### PR DESCRIPTION
**Pull request by arctouch-vivyanrodrigues**
**Description**
- [x] - Each slide should have its own indicator and the current slide indicator should be active
- [x] - During the transition between slides, the current indicator should fade out while the next indicator's slide fades in
- [x] - The animation should pause if the transition is being handled by a finger swipe
- [x] - The indicator should not be recreated every time the current slide changes

**Ticket**
AWE-6 - Animation Indicator

**Demo**

**iPhone 8:**
iphone8

**iPhone 12:**
iphone12

**Android:**
android

### CODE REVIEW

File: lib/screens/onboarding_screen.dart [hunk 1]
```diff
@@ -35,15 +35,25 @@ class _OnboardingState extends State<Onboarding> {
   ];
   final Controller controller = Controller();
   final PageController pageController = PageController(initialPage: 0);
+  late AnimationController _indicatorColorController;
 
   @override
   void initState() {
+    _indicatorColorController =
```
Comment (arctouch-victorbotamedi) [inline lib/screens/onboarding_screen.dart:42]: This controller is unnused

File: lib/components/page_index_indicator.dart [hunk 2]
```diff
@@ -1,22 +1,58 @@
...
+  final double currentPage;
+  final double page;
+  final List slides;
+
   const PageIndexIndicator({
     Key? key,
-    required this.isCurrentPage,
+    required this.currentPage,
+    required this.page,
+    required this.slides,
```
Comment (arctouch-victorbotamedi) [inline lib/components/page_index_indicator.dart:13]: slides is not used

File: lib/screens/onboarding_screen.dart (hunk 1 above)
Comment (arctouch-vivyanrodrigues) [reply lib/screens/onboarding_screen.dart]: Done!

File: lib/components/page_index_indicator.dart (hunk 2 above)
Comment (arctouch-vivyanrodrigues) [reply lib/components/page_index_indicator.dart:13]: Done!

**Review by arctouch-victorbotamedi — APPROVED** [review]

**Pull request merged**